File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.java
class Solution {
    public long solution(int w, int h) {
        
        long size = (long)w*(long)h;
        long sum = w + h;
        long gcd = gcd(w, h);
            
        long answer = size-(sum-gcd);
        return answer;
    }
    
    public long gcd(int w, int h){
        if(w%h == 0) return h;
        else return gcd(h, w%h);
    }
    
}

File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.js
function solution(w, h) {
    
    var dup = gcd(w, h);
    var size = w*h;
    var sum = w+h;
    
    var answer = size - (sum - dup);
    
    return answer;
}

function gcd(w, h){
    if(w%h == 0) return h;
    else return gcd(h, w%h);
}